common/src/main/java/eu/europeana/edm/ocr/Alto2EDMTranslator.java

```diff
@@ -0,0 +1,96 @@
+package eu.europeana.edm.ocr;
+
+import eu.europeana.edm.text.FullTextResource;
+import eu.europeana.fulltext.AnnotationType;
+import eu.europeana.edm.FullTextAnnotation;
+import eu.europeana.edm.FullTextPackage;
+import eu.europeana.edm.media.ImageDimension;
+import eu.europeana.edm.media.MediaReference;
+import eu.europeana.fulltext.alto.model.AltoPage;
+import eu.europeana.fulltext.resize.IIIFImageInfoSupport;
+import eu.europeana.fulltext.resize.ImageScale;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Iterator;
+
+/**
+ * @author Hugo Manguinhas <hugo.manguinhas@europeana.eu>
+ * @since 22 Jun 2018
+ */
+public class Alto2EDMTranslator {
+    private static final Logger LOG = LogManager.getLogger(Alto2EDMTranslator.class);
+
+    private static final String ERR_IMG = "Error obtaining image size from URL: ";
+    private static final String ERR_ALTO = "Error obtaining image size from ALTO: ";
+    private static final String INFO_SAME = "Image and Alto sizes are the same, skipping: ";
+    private static final String ERR_MIS_COORD = "Annotation is missing coordinates, skipping: ";
+
+    private boolean resize = true;
+    private IIIFImageInfoSupport support = null;
+
+    public Alto2EDMTranslator() {
+        this(false);
+    }
+
+    public Alto2EDMTranslator(boolean resize) {
+        this(resize, new IIIFImageInfoSupport());
+    }
+
+    public Alto2EDMTranslator(boolean resize, IIIFImageInfoSupport support) {
+        this.resize = resize;
+        this.support = support;
+    }
+
+    public FullTextPackage processPage(AltoPage altoPage, MediaReference ref) {
+        AnnotationsGenerator annotationsGenerator = new AnnotationsGenerator();
+        // process the fulltext annotations and resource value
+        FullTextPackage fulltext = annotationsGenerator.process(altoPage, ref);
+        if (!resize) {
+            return fulltext;
+        }
+
+        String imgURL = ref.getResourceURL();
+        ImageDimension d1 = support.getImageSize(imgURL);
+        ImageDimension d2 = altoPage.getDimension();
+        if (d1 == null) {
+            LOG.error(" {} {} ", ERR_IMG, imgURL);
+            return fulltext;
+        }
+        if (d2 == null) {
+            LOG.error("{} {} ", ERR_ALTO, imgURL);
+            return fulltext;
+        }
+        if (d1.isEquals(d2)) {
+            LOG.error(" {} {} ", INFO_SAME , imgURL);
```

Comment:
I think in these cases we just want to Log the errors and return the processed fulltext (without image resize) if the dimensions fetched are not correct or null.

Reply:
But also best to discuss with Hugo too

Reply:
better to leave it with the logging... and we need to discuss what to do about the scalling. I am thinking that the rescalling should be a separate step in the workflow that we should be able to run separately, but also in this case, logging would be the best because we dont have control over the info.json